### PR TITLE
Updated the README.md setup documentation.

### PR DESCRIPTION
**Pull request by apolloakora**
# Motivation and Context
Documentation referred to Dockerhub for image retrieval. Also laying out the python environment setup using pyenv is key to successfully running the docker-compose locally.

# What has changed
Added python setup commands and removed Dockerhub references.

# How to test?
Grab a clean MAC (uninstall python - do a make down and docker system prune) - then follow the docs to test the environment comes up as expected.

### CODE REVIEW

**Comment by apolloakora** [comment]
[Update README for docker dev and acceptance test projects for new starters](https://trello.com/c/ue1bSYwD/849-update-readme-for-docker-dev-and-acceptance-test-projects-for-new-starters)

**Comment by apolloakora** [comment]
Upgrading the python version is always a good thing. Even then though we'll still need to pin the python version because we can't guarantee the system version.

Shall I update the docs to reflect python version 3.X.X so that it is not a workaround but a placeholder for whatever version the project currently supports?

**Comment by AdamHawtin** [comment]
@apolloakora I was thinking it would be a good time to upgrade the python version used in this project to 3.7.x at least, updating the `python_version` in the pipfile should do it. That will probably require us to recreate the pipenv by running `pipenv --rm && pipenv install --dev` but then there shouldn't be any need to setup a python version older than 3.7 here and that should only be installed through pyenv. We try and exclusively use pyenv and pipenv anywhere we make use of python to avoid any of the issues that arise using system python installations.

File: README.md [hunk 1]
```diff
@@ -1,16 +1,66 @@
...
+
+## Configure Local Python Environments to Run Acceptance Tests
+
+The goal is to setup our python environments ready to run Python 3.6.10 for the app and 3.7.7 globally. This is validated with **`python -V`**.
+
+
+- **`pyenv --version`**       # check whether pyenv is installed
+- **`brew install pyenv`**    # install pyenv with brew
+- **`pyenv install 3.6.10`**  # the app needs python 3.6
+- **`pyenv shell 3.6.10`**    # this may fail with a pyenv init complaint
```
Comment (NickGrantONS) [inline README.md:25]: I think the pyenv init problem is due to not setting up pyenv in your .bash_profile or .zshrc properly... so maybe we should add instructions to do that

File: README.md [hunk 2]
```diff
@@ -1,16 +1,66 @@
...
+
+#### **`pyenv: shell integration not enabled. Run `pyenv init' for instructions.`**
+
+The above error will likely result. To fix it add this to the .zshrc (if using a z shell) or .bashrc (if using bash) so that **pyenv is always initialized**.
```
Comment (NickGrantONS) [inline README.md:29]: it's `.bash_profile` on a mac

File: README.md [hunk 3]
```diff
@@ -1,16 +1,66 @@
...
+- python -V                   # likely this will not be 3.6.10
+- **`pyenv shell 3.6.10`**    # this makes the environment 3.6.10
+- python -V                   # now it should say 3.6.10
+
+### Installing Python 3.6 in Applications
+
+If when running  **`make up`** below you get an error saying it can't find Python 3.6 on your system then you need to install from the website in addition to the above pyenv install.
+
+- visit **`https://www.python.org/downloads/mac-osx/`**
+- select download file **`python-3.6.8-macosx10.9.pkg`**
```
Comment (NickGrantONS) [inline README.md:48]: we should be using pyenv, not installing python directly onto our macs like this

File: README.md [hunk 4]
```diff
@@ -1,16 +1,45 @@
...
+
+#### Currently Supported Python Version is 3.6.10 and Machine Version is 3.7.7
+
+The goal is to setup our python environments ready to run Python 3.X.X (whaterver is currently supported). It is good practise to keep your machine version in line with the latest (today it is 3.P.Q).
+
+Validate your python versions with **`python -V`** and printenv.
+
+- **`pyenv --version`**       # check whether pyenv is installed
+- **`brew install pyenv`**    # install pyenv with brew
+- **`pyenv install 3.X.Y`**  # the app needs python 3.6
```
Comment (NickGrantONS) [inline README.md:28]: you should put `pyenv init` into your `.bash_profile` before you do this step

**Comment by NickGrantONS** [comment]
You need to merge master with this too

File: README.md [hunk 5]
```diff
@@ -1,16 +1,45 @@
...
+
+
+### Get PyEnv to install Python 3.P.Q globally
+
+The Mac nowadays likes 3.P.Q for the machine version. The goal is to make this so using **`pyenv`**. Do this in a fresh shell.
+
+- **`python -V`**             # May say something other than the latest 3.P.Q
+- **`pyenv install 3.P.Q`**   # bring down and install Python 3.P.Q
+- **`pyenv global 3.P.Q`**    # make Python 3.P.Q the global machine version
+- **`python -V`**             # now should say Python 3.P.Q
```
Comment (AdamHawtin) [inline README.md:39]: I don't think we need this separate section now, it's almost duplicating the step above

File: README.md [hunk 6]
```diff
@@ -1,16 +1,45 @@
...
+#### Currently Supported Python Version is 3.6.10 and Machine Version is 3.7.7
+
+The goal is to setup our python environments ready to run Python 3.X.X (whaterver is currently supported). It is good practise to keep your machine version in line with the latest (today it is 3.P.Q).
+
+Validate your python versions with **`python -V`** and printenv.
+
+- **`pyenv --version`**       # check whether pyenv is installed
+- **`brew install pyenv`**    # install pyenv with brew
+- **`pyenv install 3.X.Y`**  # the app needs python 3.6
+- **`pyenv shell 3.X.Y`**    # this may fail with a pyenv init complaint
```
Comment (AdamHawtin) [inline README.md:29]: I think a more repeatable way of setting it up is to set `pyenv local 3.x.y`, that way it generates a `.python-version` file in the directory and will automatically use this version whenever you're in this directory without interfereing with any other projects.

**Comment by apolloakora** [comment]
I've changed the pipfile locally to upgrade project to 3.7 - about to check in change.
I will add a note to the documentation to use `pipenv check` and `pipenv --rm` if the check fails.

The below logs occur when `make up` is run for the first time - after this everything proceeds as normal. Will reply to the pyenv local comment.

```
pipenv install --dev
Creating a virtualenv for this project…
Pipfile: /Users/apolloakora/IdeaProjects/census-rm-docker-dev/Pipfile
Using /usr/local/bin/python3.7m (3.7.7) to create virtualenv…
⠋ Creating virtual environment...created virtual environment CPython3.7.7.final.0-64 in 610ms
  creator CPython3Posix(dest=/Users/apolloakora/.local/share/virtualenvs/census-rm-docker-dev-Mxl0Dqrh, clear=False, global=False)
  seeder FromAppData(download=False, pip=latest, setuptools=latest, wheel=latest, via=copy, app_data_dir=/Users/apolloakora/Library/Application Support/virtualenv/seed-app-data/v1.0.1)
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

✔ Successfully created virtual environment! 
Virtualenv location: /Users/apolloakora/.local/share/virtualenvs/census-rm-docker-dev-Mxl0Dqrh
Installing dependencies from Pipfile.lock (a74d8c)…
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 4/4 — 00:00:04
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
pipenv run python setup_database.py
Loading .env environment variables…
```

File: README.md (hunk 6 above)
Comment (apolloakora) [reply README.md:29]: Agreed - I've placed the `.python-version` in `.gitignore` - does that seem fair? Also documenting the pyenv local option.

File: README.md (hunk 5 above)
Comment (apolloakora) [reply README.md:39]: Agreed - removing the global section.

File: README.md (hunk 4 above)
Comment (apolloakora) [reply README.md:28]: I've added pyenv init just below - if a newbie places it before they do the brew install they'll get an error whenever they open a shell.

File: README.md (hunk 2 above)
Comment (apolloakora) [reply README.md]: I've added the 3 common files .inputrc or .profile or .bash_profile for Mac users not using zshell. Yes .bashrc is a mistake - thanks for that.

File: README.md (hunk 4 above)
Comment (NickGrantONS) [reply README.md:28]: that's why I put my comment here... after the brew install

**Pull request merged**